skore/src/skore/_sklearn/_plot/metrics/confusion_matrix.py

```diff
@@ -185,54 +290,121 @@ def _compute_data_for_display(
-        cm_pred = cm / cm.sum(axis=0, keepdims=True)
-        cm_all = cm / cm.sum()
+        col_sums = cms.sum(axis=1, keepdims=True)
+        cm_pred = np.divide(cms, col_sums, where=col_sums != 0)
+
+        total_sums = cms.sum(axis=(1, 2), keepdims=True)
+        cm_all = np.divide(cms, total_sums, where=total_sums != 0)
+
+        n_thresholds = len(thresholds)
         n_classes = len(display_labels)
+        n_cells = n_classes * n_classes
+
+        true_labels = np.tile(np.repeat(display_labels, n_classes), n_thresholds)
+        pred_labels = np.tile(np.tile(display_labels, n_classes), n_thresholds)
+        threshold_values = np.repeat(thresholds, n_cells)
+
+        counts = cms.reshape(-1)
+        normalized_true_values = cm_true.reshape(-1)
+        normalized_pred_values = cm_pred.reshape(-1)
+        normalized_all_values = cm_all.reshape(-1)
 
         confusion_matrix = pd.DataFrame(
             {
-                "true_label": np.repeat(display_labels, n_classes),
-                "predicted_label": np.tile(display_labels, n_classes),
-                "count": cm.flatten(),
-                "normalized_by_true": cm_true.flatten(),
-                "normalized_by_pred": cm_pred.flatten(),
-                "normalized_by_all": cm_all.flatten(),
+                "true_label": true_labels,
+                "predicted_label": pred_labels,
+                "count": counts,
+                "normalized_by_true": normalized_true_values,
+                "normalized_by_pred": normalized_pred_values,
+                "normalized_by_all": normalized_all_values,
+                "threshold": threshold_values,
             }
         )
 
         disp = cls(
             confusion_matrix=confusion_matrix,
             display_labels=display_labels,
             report_type=report_type,
+            ml_task=ml_task,
+            pos_label=pos_label,
+            response_method=_check_response_method(
+                estimators[0], response_method
+            ).__name__,
+            thresholds=np.unique(thresholds),
         )
 
         return disp
 
-    def frame(self, normalize: Literal["true", "pred", "all"] | None = None):
-        """Return the confusion matrix as a dataframe.
+    def frame(
+        self,
+        threshold_value: float | None = None,
+    ):
+        """Return the confusion matrix as a long format dataframe.
+
+        In binary classification, the confusion matrix can be returned at various
+        decision thresholds. This is useful for understanding how the model's
+        predictions change as the decision threshold varies. If no threshold is
+        provided, the default threshold (0.5) is used.
+
+        The matrix is returned as a long format dataframe where each line represents one
+        cell of the matrix. The columns are "true_label", "predicted_label", "count",
+        "normalized_by_true", "normalized_by_pred", "normalized_by_all" and "threshold".
```

Comment:
Oh, let's align the API of plot and frame: add a `normalize` parameter. If people wants to access everything they can always request the attribute `.confusion_matrix`.